lib/lte_link_control/modules/ncellmeas.c

```diff
@@ -793,7 +821,11 @@ int ncellmeas_cancel(void)
 		err = -EFAULT;
 	}
 
-	k_sem_give(&ncellmeas_idle_sem);
+	/* Transition to idle happens when %NCELLMEAS notification is received. However, if modem
+	 * had not yet started the measurement, no notification is sent and we need a timeout to
+	 * handle it.
+	 */
+	k_work_schedule(&ncellmeas_cancel_timeout_work, K_SECONDS(1));
```

Comment:
I'm thinking if this should be 2s instead. Just because I don't know if the notification could take a bit more time when 10+ GCI cells are found and network conditions get a bit worse. It shouldn't be a problem because no the notification always comes so client knows when a new request can be made.

Reply:
Changed to 2s.